src/screens/LoginScreen.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState } from 'react';
+import { StatusBar } from 'expo-status-bar';
+import {
+  StyleSheet,
+  Text,
+  View,
+  SafeAreaView,
+  KeyboardAvoidingView,
+  Platform,
+  TouchableWithoutFeedback,
+  Keyboard
+} from 'react-native';
+import LabeledInput from '../components/LabeledInput';
+import Button from '../components/Button';
+import { Formik } from 'formik';
+import { Colors, Fonts } from '../global-styles';
+import { platformValue } from '../util/platform';
+import Alert, { AlertProps } from '../components/Alert';
+import loginAPI from '../util/loginAPI';
+import { AxiosError } from 'axios';
+
+type Credentials = {
+  nid: string;
+  password: string;
+};
+
+type ErrorObject = {
+  title: string;
+  message: string;
+  type: AlertProps['type'];
+};
+
+const LoginScreen = (): JSX.Element => {
+  const [isAlertShowing, setShowAlert] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorObject, setErrorObject] = useState<ErrorObject>({
+    title: '',
+    message: '',
+    type: undefined
+  });
+
+  // login function called when login button is pressed
+  const login = async (credentials: Credentials) => {
+    Keyboard.dismiss();
+    setIsLoading(true);
+    // seee if either nid or password field is empty, if it is display error
+    if (credentials.nid == '' || credentials.password == '') {
```

Comment:
Small nit but these should be triple equals here. As for the error, a cleaner error message would be:

```
title: Invalid Credentials
message: Your NID and password are required
```

You can also add a return from this `if` statement, this would allow you to get rid of the `else` below 👇

Reply:
done